File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    int start = 0;
    int end = 0;
    int sum = 0;
    int min = 1e9;
    vector<int> answer(2);
    while (start < sequence.size()) {
        if (sum == k) {
            if (end - start < min) {
                min = end - start;
                answer[0] = start;
                answer[1] = end - 1;
            }
            sum -= sequence[start];
            start++;
        } else if (sum > k || end >= sequence.size()) sum -= sequence[start++];
        else sum += sequence[end++];
    }
    return answer;
}

File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
       int[] answer = new int[2];
        int start = 0;
        int end = 0;
        int min = 2000000000;
        int minStart = start;
        int minEnd = start;
        int sum = 0;
        while (start < sequence.length) {
            // System.out.println("sum : " + sum);
            if (sum == k) {
                // System.out.println("same - start : " + start + ", end : " + end);
                if (min > end - start - 1) {
                    min = end - start - 1;
                    answer[0] = start;
                    answer[1] = end - 1;
                }
                sum -= sequence[start];
                start++;
            } else if (sum > k || end == sequence.length) {
                // System.out.println("over - start : " + start + ", end : " + end);
                sum -= sequence[start];
                start++;
            } else {
                // System.out.println("less - start : " + start + ", end : " + end);
                sum += sequence[end];
                end++;
            }
        }
        // answer = new int[2];
        // answer[0] = minStart;
        // answer[1] = minEnd;
    
        return answer;
    }
}